www_src/pages/project/transforms.js

```diff
@@ -0,0 +1,202 @@
+var assign = require('react/lib/Object.assign');
+var calculateSwipe = require('../../lib/swipe.js');
+
+var MAX_ZOOM = 0.8;
+var MIN_ZOOM = 0.18;
+var ZOOM_SENSITIVITY = 300;
+
+/**
+ * Directly manipulate an element's style, rather than changing it through render()
+ */
+function dangerouslySetStyle(element, style) {
+  assign(element.style, style);
+}
+
+/**
+ * utility function for constraining a value
+ */
+function constrain(v,min,max) {
+  return v<min ? min : v>max ? max : v;
+}
+
+/**
+ * Touch handler for projects
+ */
+var handleTouches = function(component) {
+  var node = component.getDOMNode();
+  var master = component.refs.bounding.getDOMNode();
+
+  // administrative values:
+  var didMove = false,
+      startX,
+      startY,
+      startDX = 0,
+      startDY = 0,
+      startDistance,
+      endX,
+      endY,
+      currentX,
+      currentY,
+      currentZoom;
+
+  // partial reset, used by handleTouchEnd
+  var resetValues = function() {
+    startX = undefined;
+    startY = undefined;
+    startDX = 0;
+    startDY = 0;
+    startDistance = undefined;
+    currentX = undefined;
+    currentY = undefined;
+    currentZoom = undefined;
+  };
+
+  // Swipe handling is supposed to kick in for... something...
+  var handleSwipe = () => {
+    // FIXME: TODO: turned off for now, in part because of Laura's comments, in
```

Comment:
not sure if we want to keep this disabled or reenable it. On a nexus 7, the swiping does pretty much nothing visually different from regular translation